doc/source/admin/data.md

```diff
@@ -0,0 +1,595 @@
+
+This document is going to describe Galaxy configuration template libraries that allow the 
+administrator to setup templates for file sources and object stores that your users may instantiate
+as they see fit. User's can instantiate multiple instances of any template, the template concept
+can apply to both file source and object store plugins, and the user interface is unified from the
+template configuration file (you as the admin do not need to explicitly declare user preferences and
+your users do not need to navigate seemingly unrelated preferences to get plugins to work).
+
+## Object Store Templates
+
+Galaxy's object store templates are configured as a YAML list of template objects. This list
+can be placed ``object_store_templates.yml`` in Galaxy configuration directory (or any path
+pointed to by the configuration option ``object_store_templates_config_file`` in ``galaxy.yml``).
+Alternatively, the  configuration can be placed directly into ``galaxy.yml`` using the
+``object_store_templates`` configuration option.
+
+A minimal object store template might look something like:
+
+```{literalinclude} ../../../lib/galaxy/objectstore/templates/examples/simple_example.yml
+:language: yaml
+```
+
+### Object Store Types
+
+#### ``disk``
+
+This is the most basic sort of object store template that just makes disk paths available to users
+for storing data. Paths can be built up from the user supplied variables, user details, supplied
+environment variables, etc.. The simple example used to demonstrate these concepts just uses
+a user supplied project name and the user's username to produce a unique path for each user
```

Comment:
Q: what happens when a user changes its username? The username is unique I assume, but can be changed.